app/main/views.py

```diff
@@ -33,7 +35,37 @@
 @main.route('/')
 def index():
     template_data = get_template_data(main, {})
-    return render_template('index.html', **template_data)
+    temporary_message = {}
+
+    try:
+        frameworks = data_api_client.find_frameworks().get('frameworks')
+        framework = get_one_framework_by_status_in_order_of_preference(
+            frameworks,
+            ['open', 'coming', 'pending']
+        )
+
+        content_loader.load_messages(framework.get('slug'), ['homepage-sidebar'])
+        temporary_message = content_loader.get_message(
+            framework.get('slug'),
+            'homepage-sidebar',
+            framework.get('status')
+        )
+
+    # if no framework is found (should never happen), ditch the message and load the page
+    except (APIError, AttributeError):
```

Comment:
If `data_api_client.find_frameworks()` returns 
```json
{ 
  "frameworks": {}
}
```
then `framework.get('slug')` will throw an error.
Don't think it's possible to happen in a real-life scenario.